subprojects/platform-play/src/main/java/org/gradle/play/internal/run/PlayApplicationRunner.java

```diff
@@ -70,7 +70,7 @@ private PlayClassloaderMonitorDeploymentDecorator(Deployment delegate, PlayRunSp
         private FileCollection collectApplicationClasspath(PlayRunSpec runSpec) {
             Set<File> applicationClasspath = Sets.newLinkedHashSet(runSpec.getChangingClasspath());
             applicationClasspath.add(runSpec.getApplicationJar());
-            return new SimpleFileCollection(applicationClasspath);
+            return ImmutableFileCollection.of(applicationClasspath);
```

Comment:
I'm not saying we should spend a ton of time optimizing this, but we could use an `ImmutableSet.builder()` instead of the `LinkedHashSet` here and already pass an `ImmutableSet`.